ledger/metrics/src/main/scala/com/daml/metrics/Spans.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics
+
+import akka.Done
+import akka.stream.scaladsl.{Keep, Source}
+import io.opentelemetry.common.Attributes
+import io.opentelemetry.trace.TracingContextUtils
+
+import scala.concurrent.Future
+
+object Spans {
+  def attachEventsToCurrentSpan[Out, Mat](
+      extractEvents: Out => Iterable[Event],
+      source: => Source[Out, Mat]): Source[Out, Mat] = {
+    source
```

Comment:
This feels more complicated than it should be. Does `source.alsoTo(Sink.foreach(item => ...))` work or does it change the type?

Reply:
Good shout! I didn't know about `alsoTo`. I'll look into that and `wireTap`.

Reply:
Switched to `wireTap`, which simplifies this quite a bit.